src/sentry/static/sentry/app/views/events/searchBar.tsx

```diff
@@ -25,25 +26,34 @@ const FIELD_TAGS = Object.fromEntries(
   Object.keys(FIELDS).map(item => [item, {key: item, name: item}])
 );
 
-class SearchBar extends React.PureComponent {
-  static propTypes = {
+type SearchBarProps = Omit<React.ComponentProps<typeof SmartSearchBar>, 'tags'> & {
+  api: Client;
+  organization: Organization;
+  tags: {[key: string]: Tag};
+  omitTags?: string[];
+  projectIds?: number[] | Readonly<number[]>;
+  fields?: Readonly<Field[]>;
```

Comment:
Do you know why we need these as `Readonly`?

Reply:
I'm mostly following from how the component is consumed, as I figured if we want searchBar to take a mutable field array in the future we can add the functionality to cast from readonly -> mutable and allow both types in props. As for why it's `Readonly` in the first place, `fields` come from EventView in all instances of fields being passed to `SearchBar` I could find (performance/landing, transactionSummary/content, eventsV2/results). EventView has the more restrictive `Readonly<Field[]>`.